lrpar/src/lib/lex.rs

```diff
@@ -52,12 +50,12 @@ pub trait Lexer<StorageT: Hash + PrimInt + Unsigned> {
     }
 }
 
-/// A `Lexeme` represents a segment of the user's input that conforms to a known type. All lexemes
-/// have a starting position in the user's input: lexemes that result from error recovery, however,
-/// do not have a length (or, therefore, an end). This allows us to differentiate between lexemes
-/// that are always of zero length (which are required in some grammars) from lexemes that result
-/// from error recovery (where an error recovery algorithm can know the type that a lexeme should
-/// have been, but can't know what its contents should have been).
+/// A `Lexeme` represents a segment of the user's input that conforms to a known type. Note that
+/// even if the type of a lexeme seemingly requires it to have `len() > 0` (e.g. integers might
+/// match the regular expressions `[0-9]+`), error recovery might cause a lexeme to have a length
+/// 0. Users can detect the difference between a lexeme with an intentionally zero length from
```

Comment:
`of 0`?

Reply:
Fixed in https://github.com/softdevteam/grmtools/pull/125/commits/0c3ab45f2f5ac5b5d9352c022a05a5def819864e.